File: Basic/JS/1008.js
/**
 * @author: kurumi
 * @date: 2025-09-07
 * @description: 1008 数组元素循环右移问题
 */
const rl = require("readline").createInterface({input: process.stdin});
const iter = rl[Symbol.asyncIterator]();
const readline = async () => (await iter.next()).value;

void (async function() {
  const [n, m] = (await readline()).split(" ").map(Number);
  const nums = (await readline()).split(" ");
  const res = new Array();
  for (let i = n - m % n; i < n; i++) res.push(nums[i]);
  for (let i = 0; i < n - m % n; i++) res.push(nums[i]);
  console.log(res.join(" "));
})();

File: Basic/Python/1008.py
'''
@Author: kurumi
@Date: 2025-09-07
@Description: 1008 数组元素循环右移问题
'''
from typing import List

def main():
    n, m = map(int, input().split())
    nums: List[int] = list(map(str, input().split()))
    res: List[int] = []
    for i in range(n - m % n, n):
        res.append(nums[i])
    for i in range(n - m % n):
        res.append(nums[i])
    print(" ".join(res))

if __name__ == "__main__":
    main()